### PR TITLE
release: v1.31.0

### PR DESCRIPTION
**Pull request by abernix**
> **Note**
> **This particular PR must be true-merged to `main`.**

* This PR is only ready to review when it is marked as "Ready for Review".  It represents the merge to the `main` branch of an upcoming release (version number in the title).
* It will act as a staging branch until we are ready to finalize the release.
* We may cut any number of alpha and release candidate (RC) versions off this branch prior to formalizing it. 
* This PR is **primarily a merge commit**, so reviewing every individual commit shown below is **not necessary** since those have been reviewed in their own PR.  However, things important to review on this PR **once it's marked "Ready for Review"**:
	- Does this PR target the right branch? (usually, `main`)
	- Are the appropriate **version bumps** and **release note edits** in the end of the commit list (or within the last few commits).  In other words, "Did the 'release prep' PR actually land on this branch?"
	- If those things look good, this PR is good to merge!

### CODE REVIEW

File: .changesets/feat_geal_plan_cache_warmup_doc.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+### Query plan cache warm-up improvements ([Issue #3704](https://github.com/apollographql/router/issues/3704))
+
+The `warm_up_queries` option enables quicker schema updates by precomputing query plans for your most used cached queries and your persisted queries. When a new schema is loaded, a precomputed query plan for it may already be in in-memory cache.
+
+We made a series of improvements to this feature to make it more usable:
+* It is now active by default and warms up the cache with the 30% most used queries of the previous cache. The amount is still configurable, and it can be deactivated by setting it to 0.
+* We added new metrics to track the time spent loading a new schema and planning queries in the warm-up phase. You can also measure the query plan cache usage, to know how many entries are used, and the cache hit rate, both for the in memory cache and the distributed cache.
```
Comment (lrlna) [inline .changesets/feat_geal_plan_cache_warmup_doc.md:7]: @Geal is this part of the work Benjamin did [in this changeset](https://github.com/apollographql/router/pull/3861/files#diff-0a6b9559d47c40fd09b21288e986861d161480485228474398369dc2535257e6), or is that something entirely separate?

Comment (Geal) [reply .changesets/feat_geal_plan_cache_warmup_doc.md:7]: the work of Benjamin is part of this changeset. #3704 is an umbrella issue for the query planner improvements. So the changesets should be merged

Comment (lrlna) [reply .changesets/feat_geal_plan_cache_warmup_doc.md:7]: ty!

File: .changesets/feat_bnjjj_fix_3621.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+### feat(redis): add configuration to set the timeout ([Issue #3621](https://github.com/apollographql/router/issues/3621))
+
+It adds a configuration to set another timeout than the default one (2ms) for redis requests. It also change the default timeout to 2ms (previously set to 1ms)
```
Comment (chandrikas) [inline .changesets/feat_bnjjj_fix_3621.md:3]:
```suggestion
It's now possible to override the default timeout for redis requests with a new configuration parameter. In this change, we've also increased the default timeout to 2ms (up from 1ms previously) 
```

**Review by BrynCooke — APPROVED** [review]

**Review by garypen — APPROVED** [review]

**Pull request merged**